pkg/kubectl/cmd/taint_test.go

```diff
@@ -0,0 +1,299 @@
+		oldTaints   []api.Taint
+		newTaints   []api.Taint
+		args        []string
+		expectFatal bool
+		expectTaint bool
+	}{
+		// success cases
+		{
+			description: "taints a node with effect NoSchedule",
+			newTaints: []api.Taint{{
+				Key:    "foo",
+				Value:  "bar",
+				Effect: "NoSchedule",
+			}},
+			args:        []string{"node", "node-name", "foo=bar:NoSchedule"},
+			expectFatal: false,
+			expectTaint: true,
+		},
+		{
+			description: "taints a node with effect PreferNoSchedule",
+			newTaints: []api.Taint{{
+				Key:    "foo",
+				Value:  "bar",
+				Effect: "PreferNoSchedule",
+			}},
+			args:        []string{"node", "node-name", "foo=bar:PreferNoSchedule"},
+			expectFatal: false,
+			expectTaint: true,
+		},
+		{
+			description: "update an existing taint on the node, change the effect from NoSchedule to PreferNoSchedule",
+			oldTaints: []api.Taint{{
+				Key:    "foo",
+				Value:  "bar",
+				Effect: "NoSchedule",
+			}},
+			newTaints: []api.Taint{{
+				Key:    "foo",
+				Value:  "bar",
+				Effect: "PreferNoSchedule",
+			}},
+			args:        []string{"node", "node-name", "foo=bar:PreferNoSchedule", "--overwrite"},
+			expectFatal: false,
+			expectTaint: true,
+		},
+		{
+			description: "taints a node with two taints",
+			newTaints: []api.Taint{{
+				Key:    "dedicated",
+				Value:  "namespaceA",
+				Effect: "NoSchedule",
+			}, {
+				Key:    "foo",
+				Value:  "bar",
+				Effect: "PreferNoSchedule",
+			}},
+			args:        []string{"node", "node-name", "dedicated=namespaceA:NoSchedule", "foo=bar:PreferNoSchedule"},
+			expectFatal: false,
+			expectTaint: true,
+		},
+		{
+			description: "node has two taints, remove one of them",
+			oldTaints: []api.Taint{{
+				Key:    "dedicated",
+				Value:  "namespaceA",
+				Effect: "NoSchedule",
+			}, {
+				Key:    "foo",
+				Value:  "bar",
+				Effect: "PreferNoSchedule",
+			}},
+			newTaints: []api.Taint{{
+				Key:    "foo",
+				Value:  "bar",
+				Effect: "PreferNoSchedule",
+			}},
+			args:        []string{"node", "node-name", "dedicated-"},
+			expectFatal: false,
+			expectTaint: true,
+		},
+		{
+			description: "node has two taints, update one of them and remove the other",
+			oldTaints: []api.Taint{{
+				Key:    "dedicated",
+				Value:  "namespaceA",
+				Effect: "NoSchedule",
+			}, {
+				Key:    "foo",
+				Value:  "bar",
+				Effect: "PreferNoSchedule",
+			}},
+			newTaints: []api.Taint{{
+				Key:    "foo",
+				Value:  "bar",
+				Effect: "NoSchedule",
+			}},
+			args:        []string{"node", "node-name", "dedicated-", "foo=bar:NoSchedule", "--overwrite"},
+			expectFatal: false,
+			expectTaint: true,
+		},
+
+		// error cases
+		{
+			description: "invalid taint key",
+			args:        []string{"node", "node-name", "nospecialchars^@=banana:NoSchedule"},
+			expectFatal: true,
+			expectTaint: false,
+		},
+		{
+			description: "invalid taint effect",
+			args:        []string{"node", "node-name", "nospecialchars^@=banana:NoExcute"},
```

Comment:
Probably would be better if you use a key that is valid, so you are only testing invalid taint effect.